services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Utils.kt

```diff
@@ -33,8 +36,15 @@ fun currentUserId(): String = SecurityContextHolder.getContext().authentication.
 fun BusinessId.isValidBusinessId(): Boolean {
     logger.info { "Verifying businessId: $this" }
 
-    if (length != 9 || this[7] != '-') {
-        return false.also { logger.warn { "Invalid format." } }
+    if (length != 9 || substring(0, 7).any { !it.isDigit() } || this[7] != '-') {
```

Comment:
This check is getting complex enough that a regex would be a simplification. The same regex could be used to split the id to the sequence number and check digit parts.

```kotlin
    val businessIdRegex = "^(\\d{7})-(\\d)\$".toRegex() // Can be at class level

    val (sequenceNumber, checkDigit) =
        businessIdRegex.find(this)?.destructured
            ?: return false.also { logger.warn { "Invalid format." } }
```